### PR TITLE
HTCONDOR-2650 Add htcondor.exe, a python C API program for the noun/v…

### PR DESCRIPTION
**Pull request by johnkn**
…erb tool on Windows.

<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [x] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed  (documentation [build logs](https://readthedocs.org/projects/htcondor/builds/))
- [x] Check for version history, if needed
- [x] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by Todd-L-Miller** [comment]
Sorry, apparently GitHub didn't actually _post_ any of the comments I made when I made them.  I will go through and update according to our conversation about them.

File: docs/version-history/lts-versions-24-0.rst [hunk 1]
```diff
@@ -27,6 +27,11 @@ New Features:
 
 Bugs Fixed:
 
+- On Windows the :tool:`htcondor` tool now uses the python API to try and the
+  launch the python interpretor.  This will fail with a message
+  box about installing python if python 3.9 is not in the path.
```
[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply docs/version-history/lts-versions-24-0.rst:32]: According to TJ, the distributions _include_ the library, but whatever we're currently doing in CMake to _find_ the library doesn't work.  The blocker for moving to the limited API is actually that the version 1 bindings require the version-specific API, and the `htcondor` CLI currently uses version 1; see [HTCONDOR-2623](https://opensciencegrid.atlassian.net/browse/HTCONDOR-2623).

(Please resolve this conversation once you've seen this.)

File: src/condor_tools/htcondor.c [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#endif
+
+	// "import sys"
+	// we use AddModule rather than ImportModule here because sys is always loaded.
+	//PyObject * sys = PyImport_AddModule("sys");
+
+	// "from htcondor_cli.cli import main"
+	PyObject * cli_import_list = PyList_New(1);
+	PyList_Append(cli_import_list, PyUnicode_FromString("main"));
+	PyObject * cli = PyImport_ImportModuleLevel("htcondor_cli.cli", NULL, NULL, cli_import_list, 0);
```
[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply src/condor_tools/htcondor.c:102]: Sounds like that was, in fact, the case.

(Please resolve this conversation once you've seen this.)

File: src/condor_tools/htcondor.c [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	} else {
+		PyObject * result = PyObject_CallObject(fnmain, NULL);
+		// unpack the main return value
+		ret = (int)PyLong_AsLong(result);
+	}
+
+	// free stuff
+	Py_DECREF(fnmain);
+	Py_DECREF(cli);
+	//Py_DECREF(sys);
```
[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply src/condor_tools/htcondor.c:119]: Where "this series of stuff" was probably referring to the commented-out code dealing with `sys`.

**Review by Todd-L-Miller — COMMENTED** [review]
Generally looks good, but before I approve -- would it be hard to write just a smoke-test (even if we can't get the existing htcondor tests working on Windows in a reasonable amount of time)?

File: src/condor_tools/htcondor.c [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+#if defined(_MSC_VER)
+ #pragma push_macro("PLATFORM")
+ #undef PLATFORM
+
+ #pragma push_macro("_DEBUG")
+ #undef _DEBUG
+#endif
+
+// if you enable limited api, then you have to also change the cmake to link with python3.lib
+//#define Py_LIMITED_API 0x03090000
```
[the remark this answers is not in the document]
Comment (johnkn) [reply src/condor_tools/htcondor.c:30]: That will be the one that we build with, not neceessarily the minimum we need for the code to work.

[the remark this answers is not in the document]
Comment (Todd-L-Miller) [reply src/condor_tools/htcondor.c:30]: To be clear, whatever value you put in there will be the floor for the versions that the limited API will run with; the version 2 bindings currently (or will shortly) use `0x03030000` for stupid technical reasons.  Unless you need something from 3.9, when we get this working, we should use that string.

**Comment by Todd-L-Miller** [comment]
TJ says that even a smoke-test would take too long to write.

**Review by Todd-L-Miller — APPROVED** [review]

**Pull request merged**